src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -160,20 +160,22 @@ float addTempRow(
       int curRowIndex,
       Map<String, RowBufferStats> statsMap,
       Set<String> formattedInputColumnNames) {
-    return addRow(row, this::writeRow, statsMap, formattedInputColumnNames);
+    return addRow(row, curRowIndex, this::writeRow, statsMap, formattedInputColumnNames);
```

Comment:
The `curRowIndex` logic doesn't look correct and may be used for something else, if you see `insertRows` in AbstractRowBuffer, it's the row count instead of the original row index